spec/features/consumer/shopping/shopping_spec.rb

```diff
@@ -232,6 +232,28 @@
           expect(page).not_to have_content variant3.display_name
         end
       end
+
+      context "when the distributor has no available payment/shipping methods" do
+        before do
+          distributor.update_attributes shipping_methods: [], payment_methods: []
+        end
+
+        # Display only shops are a very useful hack that is described in the user guide
+        it "still renders a display only shop" do
+          visit shop_path
+          expect(page).to have_content product.name
+
+          # Add product to cart
+          fill_in "variants[#{variant.id}]", with: '1'
+          wait_for_debounce
+          expect(page).to have_in_cart product.name
+          wait_until { !cart_dirty }
+
+          # Try to go to cart
+          visit main_app.cart_path
+          expect(page).to have_content "The hub you have selected is temporarily closed for orders. Please try again later."
+        end
```

Comment:
can't we check the feature from a controller spec? if this is about showing the products form when there are no open OCs it should. You know, the "inverted" testing pyramid we have always concerned me.

Reply:
not in this case, this is really a feature. It would be somewhat testable in a view spec but I think we should go with feature spec to make sure it works as described in the user guide.